src/Lecture1.hs

```diff
@@ -1,126 +1,145 @@
-73
-
-The string contains only spaces and/or numbers.
--}
-strSum str = error "TODO"
-
-{- | Write a function that takes a number and a list of numbers and
-returns a string, saying how many elements of the list are strictly
-greater than the given number and strictly lower.
-
->>> lowerAndGreater 3 [1 .. 9]
-"3 is greater than 2 elements and lower than 6 elements"
-
-Explanation: the list [1 .. 9] contains 9 elements: [1, 2, 3, 4, 5, 6, 7, 8, 9]
-The given number 3 is greater than 2 elements (1 and 2)
-and lower than 6 elements (4, 5, 6, 7, 8 and 9).
-
-🕯 HINT: Use recursion to implement this function.
--}
-lowerAndGreater n list = error "TODO"
+lastDigit :: Integral a => a -> a
+lastDigit n = abs n `mod` 10
+
+-- | Write a function that takes three numbers and returns the
+-- difference between the biggest number and the smallest one.
+--
+-- >>> minmax 7 1 4
+-- 6
+--
+-- Explanation: @minmax 7 1 4@ returns 6 because 7 is the biggest number
+-- and 1 is the smallest, and 7 - 1 = 6.
+--
+-- Try to use local variables (either let-in or where) to implement this
+-- function.
+minmax :: (Num a, Ord a) => a -> a -> a -> a
+minmax x y z =
+  let l = sort [x, y, z]
+   in last l - head l
+
+-- | Implement a function that takes a string, start and end positions
+-- and returns a substring of a given string from the start position to
+-- the end (including).
+--
+-- >>> subString 3 7 "Hello, world!"
+-- "lo, w"
+--
+-- >>> subString 10 5 "Some very long String"
+-- ""
+--
+-- This function can accept negative start and end position. Negative
+-- start position can be considered as zero (e.g. substring from the
+-- first character) and negative end position should result in an empty
+-- string.
+subString :: Int -> Int -> [a] -> [a]
+subString start end str =
+  let start' = max start 0
+   in if end < 0
+        then []
+        else take (end - start' + 1) (drop start' str)
+
+-- | Write a function that takes a String — space separated numbers,
+-- and finds a sum of the numbers inside this string.
+--
+-- >>> strSum "100    -42  15"
+-- 73
+--
+-- The string contains only spaces and/or numbers.
+strSum :: (Num a, Read a) => String -> a
+strSum = sum . map read . words
+
+-- | Write a function that takes a number and a list of numbers and
+-- returns a string, saying how many elements of the list are strictly
+-- greater than the given number and strictly lower.
+--
+-- >>> lowerAndGreater 3 [1 .. 9]
+-- "3 is greater than 2 elements and lower than 6 elements"
+--
+-- Explanation: the list [1 .. 9] contains 9 elements: [1, 2, 3, 4, 5, 6, 7, 8, 9]
+-- The given number 3 is greater than 2 elements (1 and 2)
+-- and lower than 6 elements (4, 5, 6, 7, 8 and 9).
+--
+-- 🕯 HINT: Use recursion to implement this function.
+lowerAndGreater :: (Show a, Ord a) => a -> [a] -> [Char]
+lowerAndGreater n list =
+  show n ++ " is greater than " ++ show gt
+    ++ " elements and lower than "
+    ++ show lw
+    ++ " elements"
+  where
+    (gt, lw) = go list 0 0
```

Comment:
I see a GHC warning here about defaulting polymorphic value of type `Num` to `Integer`. You can remove this warning by specifying types of those values explicitly: 
```suggestion
    gt, lw :: Int
    (gt, lw) = go list 0 0
```